src/ReducedModelControllers/src/CentroidalMPC.cpp

```diff
@@ -0,0 +1,1341 @@
+        ok = ok && frictionCone.initialize(ptr);
+
+        getOptionalParameter(ptr, "linear_solver", this->optiSettings.ipoptLinearSolver);
+        getOptionalParameter(ptr, "ipopt_tolerance", this->optiSettings.ipoptTolerance);
+        getOptionalParameter(ptr, "ipopt_max_iteration", this->optiSettings.ipoptMaxIteration);
+        getOptionalParameter(ptr, "solver_verbosity", this->optiSettings.solverVerbosity);
+        getOptionalParameter(ptr, "is_warm_start_enabled", this->optiSettings.isWarmStartEnabled);
+        getOptionalParameter(ptr, "is_cse_enabled", this->optiSettings.isCseEnabled);
+
+        return ok;
+    }
+
+    casadi::Function ode()
+    {
+        // Convert DiscreteGeometryContact into a casadiContact object
+        std::map<std::string, CasadiContact> casadiContacts;
+
+        for (const auto& [key, contact] : this->output.contacts)
+        {
+            CasadiContact temp(key);
+            temp = contact;
+            auto [contactIt, outcome] = casadiContacts.emplace(key, temp);
+        }
+
+        // we assume mass equal to 1
+        constexpr double mass = 1;
+
+        casadi::MX com = casadi::MX::sym("com_in", 3);
+        casadi::MX dcom = casadi::MX::sym("dcom_in", 3);
+        casadi::MX angularMomentum = casadi::MX::sym("angular_momentum_in", 3);
+
+        casadi::MX externalForce = casadi::MX::sym("external_force", 3);
+        casadi::MX externalTorque = casadi::MX::sym("external_torque", 3);
+
+        casadi::MX ddcom = casadi::MX::sym("ddcom", 3);
+        casadi::MX angularMomentumDerivative = casadi::MX::sym("angular_momentum_derivative", 3);
+
+        casadi::DM gravity = casadi::DM::zeros(3);
+        gravity(2) = -BipedalLocomotion::Math::StandardAccelerationOfGravitation;
+
+        ddcom = gravity + externalForce / mass;
+        angularMomentumDerivative = externalTorque;
+
+        std::vector<casadi::MX> input;
+        input.push_back(externalForce);
+        input.push_back(externalTorque);
+        input.push_back(com);
+        input.push_back(dcom);
+        input.push_back(angularMomentum);
+
+        for (const auto& [key, contact] : casadiContacts)
+        {
+            input.push_back(contact.position);
+            input.push_back(contact.orientation);
+            input.push_back(contact.isEnabled);
+            input.push_back(contact.linearVelocity);
+
+            for (const auto& corner : contact.corners)
+            {
+                using namespace casadi;
+                ddcom += contact.isEnabled / mass * corner.force;
+                angularMomentumDerivative
+                    += contact.isEnabled
+                       * MX::cross(MX::mtimes(MX::reshape(contact.orientation, 3, 3),
+                                              corner.position)
+                                       + contact.position - com,
+                                   corner.force);
+
+                input.push_back(corner.force);
+            }
+        }
```

Comment:
The problem here is that the variables defined here, for instance, the com, contacts e are local. So there is no way to force the serialization since casadi expects a std::vector of casadi::MX with a given size. So there is no way to prevent the user to invert the com position and its velocity. Still, this is the same as having a function that takes two `Eigen::Vector3d`. No one prevents inverting the input. 

A possible solution is to design a layer on top of casadi that ensure the serialization, still, this is not the scope of this PR.